src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyInlineCompletionProvider.kt

```diff
@@ -0,0 +1,206 @@
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.AutocompleteParams
+import com.sourcegraph.cody.agent.protocol.AutocompleteResult
+import com.sourcegraph.cody.agent.protocol.AutocompleteTriggerKind
+import com.sourcegraph.cody.agent.protocol.ErrorCode
+import com.sourcegraph.cody.agent.protocol.ErrorCodeUtils.toErrorCode
+import com.sourcegraph.cody.agent.protocol.ProtocolTextDocument.Companion.uriFor
+import com.sourcegraph.cody.agent.protocol.RateLimitError.Companion.toRateLimitError
+import com.sourcegraph.cody.agent.protocol.SelectedCompletionInfo
+import com.sourcegraph.cody.agent.protocol_extensions.Position
+import com.sourcegraph.cody.agent.protocol_generated.Position
+import com.sourcegraph.cody.agent.protocol_generated.Range
+import com.sourcegraph.cody.ignore.ActionInIgnoredFileNotification
+import com.sourcegraph.cody.ignore.IgnoreOracle
+import com.sourcegraph.cody.ignore.IgnorePolicy
+import com.sourcegraph.cody.statusbar.CodyStatus
+import com.sourcegraph.cody.statusbar.CodyStatusService.Companion.notifyApplication
+import com.sourcegraph.cody.statusbar.CodyStatusService.Companion.resetApplication
+import com.sourcegraph.cody.vscode.CancellationToken
+import com.sourcegraph.cody.vscode.InlineCompletionTriggerKind
+import com.sourcegraph.cody.vscode.IntelliJTextDocument
+import com.sourcegraph.common.UpgradeToCodyProNotification
+import com.sourcegraph.config.ConfigUtil
+import com.sourcegraph.utils.CodyEditorUtil.isImplicitAutocompleteEnabledForEditor
+import java.util.concurrent.CancellationException
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+import org.eclipse.lsp4j.jsonrpc.ResponseErrorException
+
+class CodyInlineCompletionProvider : InlineCompletionProvider {
+  private val logger = Logger.getInstance(CodyInlineCompletionProvider::class.java)
+  private val currentJob = AtomicReference(CancellationToken())
+
+  override suspend fun getProposals(
+      request: InlineCompletionRequest
+  ): List<InlineCompletionElement> {
+    val editor = request.editor
+    val project = editor.project ?: return emptyList()
+    val enabled = isImplicitAutocompleteEnabledForEditor(editor)
+    if (!enabled) {
+      return emptyList()
+    }
+
+    // Fetch autocompletions using existing logic
+
+    cancelCurrentJob(project)
+    val cancellationToken = CancellationToken()
+    currentJob.set(cancellationToken)
+
+    val completions =
+        fetchCompletions(project, editor, InlineCompletionTriggerKind.AUTOMATIC, cancellationToken)
+            .get() ?: return emptyList()
+
+    // Create a list to store InlineCompletionElement objects
+    val result = mutableListOf<InlineCompletionElement>()
+
+    // Add completions to the result
+    completions.items.forEach { completion ->
+      result.add(InlineCompletionElement(completion.insertText))
```

Comment:
The problem is that `InlineCompletionElement` in platform version 232 does not support offsets. Completions need to be trimmed otherwise they look like this:
<img width="873" alt="image" src="https://github.com/user-attachments/assets/dac00e8c-446b-47ec-bab6-5cf74b4d9ff4">